### DISCUSSION TITLE
Cache improvements ideas

### DISCUSSION THREAD

**Discussion by GeertvanHorrik** [cake-build/cake#3871] [Ideas]
I am experimenting with the new cache feature and so far it's working out really well. We use this to cache since we run the same script in several stages, and each time compilation can take up to 45 seconds so really happy with these improvements.

To share the cache between repositories, there are 2 possibilities:

1. Use environment variables that are process specific (e.g. %temp%/cake-buid/cache/%repositoryslug%)
2. Instead of using build.cake.dll, why not use the *hash* as the filename. In this case, the cache can always be re-used between repositories but also never have any collisions with other versions?

I think implementing 2 could benefit others as well (and happy to help out here). The only downside I see is that there is no "clean-up" of the cache. But in such case, cake could always clean up files older than say 1 month to keep the cache fairly clean.

**Comment by pascalberger** [comment]
@GeertvanHorrik We did some improvements to the caching with #3881 for next release of Cake which should solve your issue

**Reply by GeertvanHorrik** [reply]
Thank you!
